src/librbd/ImageCtx.h

```diff
@@ -158,6 +158,7 @@ namespace librbd {
     EventSocket event_socket;
 
     ContextWQ *op_work_queue;
+    ContextWQ *pcache_op_work_queue; //chendi: used for multi process cache op
```

Comment:
Since the cache will require its own thread-pool for file IO (so that it doesn't block other operations when doing blocking IO), the thread-pool and associated work queue should be owned / managed by the cache instance. They can still re-use the singleton concept since you might have multiple layers of parents being cached and we wouldn't want to start a cache IO thread pool for each image in the clone hierarchy.